src/github_utils.py

```diff
@@ -19,111 +20,107 @@ def get_pull_request_diff():
+            comment['end_line'] <= hunk['end']):
+            return {
+                'path': comment['path'],
+                'start_line': comment['start_line'],
+                'line': comment['end_line'],
+                'body': comment['body'],
+                'start_side': 'RIGHT',
+                'side': 'RIGHT'
+            }
+    print(f"Invalid range: {comment['path']} {comment['start_line']}-{comment['end_line']}")
+    return None
 
 
 def post_review_comment(comments, diff_content):
-    """Post validated review comments to GitHub"""
     g = Github(os.getenv('GIT_TOKEN'))
-
-    # Load GitHub event data
     event_path = os.getenv('GITHUB_EVENT_PATH')
+    
     with open(event_path, 'r') as f:
         event_data = json.load(f)
-
-    # Get repository and PR details
+    
     repo = g.get_repo(event_data['repository']['full_name'])
-    pr_number = event_data['pull_request']['number']
-    pull_request = repo.get_pull(pr_number)
+    pr = repo.get_pull(event_data['pull_request']['number'])
     head_sha = event_data['pull_request']['head']['sha']
-
-    # Get valid lines from diff
-    valid_lines = get_valid_lines(diff_content)
     
-    # Prepare filtered comments
+    hunks = get_valid_hunks(diff_content)
     filtered_comments = []
+    
     for comment in comments:
-
-        # Ensure we're dealing with a dictionary
-        if not isinstance(comment, dict):
-            print(f"Skipping invalid comment type: {type(comment)}")
-            continue
-
-        try:
-            file_path = comment['path']
-            line_num = comment.get('line', 1)
-            
-            # Validate against actual diff
-            if file_path in valid_lines and line_num in valid_lines[file_path]:
-                filtered_comments.append({
-                    "path": file_path,
-                    "position": line_num,  # GitHub expects 'position' not 'line'
-                    "body": comment['body']
-                })
-            else:
-                print(f"Skipping invalid comment - File: {file_path}, Line: {line_num}")
-                
-        except KeyError as e:
-            print(f"Skipping malformed comment: {comment} - Missing key: {e}")
-
-    if not filtered_comments:
-        print("No valid comments to post after filtering")
+        validated = validate_comment(comment, hunks)
+        if validated:
+            filtered_comments.append(validated)
+        else:
+            print(f"Skipping invalid: {comment['path']} {comment['start_line']}-{comment['end_line']}")
+    
+    valid_comments = []
+    for comment in filtered_comments:
+        if (comment['start_line'] <= comment['line'] and  # GitHub API requirement
+            comment['path'] in [h['file'] for h in hunks]):
+            valid_comments.append(comment)
+    
+    if not valid_comments:
+        print("No valid comments after final validation")
         return
 
+    # for a in valid_comments:
+    #     print(a)
```

Comment:
Clarifies the purpose of validating the comment
```suggestion
|
```
N/A